### PR TITLE
reference, toc: add a list of pushdown operators

### PR DESCRIPTION
**Pull request by anotherrachel**
### What is changed, added or deleted? 
This PR adds a doc to introduces a list of operators that can be pushed down to TiKV, and updates the related `TOC.md` and `reference.md`.

### What is the related PR or file link(s)? 
https://github.com/pingcap/docs-cn/pull/1802

### Which version does your change affect? 
dev
I'll update v3.0 when these changes are approved.

### CODE REVIEW

**Comment by anotherrachel** [comment]
@fzhedu @yikeke PTAL, thanks!

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
+---
+title: List of Operators for Pushdown
+summary: This document introduces a list of operators that can be pushed down to TiKV, the related operations of which are also provided.
+category: reference
+---
+
+# List of Operators for Pushdown
+
+When TiDB reads data from TiKV, TiDB tries to push down some expressions to TiKV to be processed thus reducing the amount of data transferred and the computational pressure on a single TiDB node. This document introduces operators that TiDB supports for pushdown and how to blacklist specific operators.
```
Comment (fzhedu) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]: Can the `pushdown` be used as a noun？
This document introduces which TiDB operators can be pushed down to TiKV, and how to disable this function by putting specific operators into a blacklist.

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+| [JSON Functions](/dev/reference/sql/functions-and-operators/json-functions.md) | [JSON_TYPE(json_val)][json_type],<br> [JSON_EXTRACT(json_doc, path[, path] ...)][json_extract],<br> [JSON_UNQUOTE(json_val)][json_unquote],<br> [JSON_OBJECT(key, val[, key, val] ...)][json_object],<br> [JSON_ARRAY([val[, val] ...])][json_array],<br> [JSON_MERGE(json_doc, json_doc[, json_doc] ...)][json_merge],<br> [JSON_SET(json_doc, path, val[, path, val] ...)][json_set],<br> [JSON_INSERT(json_doc, path, val[, path, val] ...)][json_insert],<br> [JSON_REPLACE(json_doc, path, val[, path, val] ...)][json_replace],<br> [JSON_REMOVE(json_doc, path[, path] ...)][json_remove] |
+| [Date and Time Functions](/dev/reference/sql/functions-and-operators/date-and-time-functions.md) | [`DATE_FORMAT()`](https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_date-format)  |
+
+## Blacklist specific operators
+
+If unexpected behavior occurs in the calculation of the function due to pushdown operators, you can quickly restore it by blacklisting specific operators. Specifically, you can prohibit operators from being pushing down by putting it on the blacklist `mysql.expr_pushdown_blacklist`.
```
Comment (fzhedu) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]: If unexpected behavior occurs during the calculation of functions due to pushed down operators, you can quickly restore the execution by blacklisting specific operators. Specifically, you can prohibit specific operators from being pushed down by putting it into the blacklist ....

File: dev/reference/sql/functions-and-operators/reference.md [hunk 3]
```diff
@@ -10,4 +10,4 @@ The usage of the functions and operators in TiDB is similar to MySQL. See [Funct
 
 In SQL statements, expressions can be used on the `ORDER BY` and `HAVING` clauses of the `SELECT` statement, the `WHERE` clause of `SELECT`/`DELETE`/`UPDATE` statements, and `SET` statements.
 
-You can write expressions using literals, column names, NULL, built-in functions, operators and so on.
+You can write expressions using literals, column names, NULL, built-in functions, operators and so on. For operators that TiDB supports for pushdown to TiKV, see [List of Operators for Pushdown](/dev/reference/sql/functions-and-operators/expressions-pushed-down.md).
```
Comment (fzhedu) [inline dev/reference/sql/functions-and-operators/reference.md]: For TiDB operators that can be pushed down to TiKV, see...

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+title: List of Operators for Pushdown
+summary: This document introduces a list of operators that can be pushed down to TiKV, the related operations of which are also provided.
+category: reference
+---
+
+# List of Operators for Pushdown
+
+When TiDB reads data from TiKV, TiDB tries to push down some expressions to TiKV to be processed thus reducing the amount of data transferred and the computational pressure on a single TiDB node. This document introduces operators that TiDB supports for pushdown and how to prohibit specific operators from being pushed down by adding them to the blacklist.
+
+## List of operators for pushdown
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
## Supported pushdown expressions
```

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+| Type | Operators |
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
| Expression Type | Operations |
```

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+| [JSON Functions](/dev/reference/sql/functions-and-operators/json-functions.md) | [JSON_TYPE(json_val)][json_type],<br> [JSON_EXTRACT(json_doc, path[, path] ...)][json_extract],<br> [JSON_UNQUOTE(json_val)][json_unquote],<br> [JSON_OBJECT(key, val[, key, val] ...)][json_object],<br> [JSON_ARRAY([val[, val] ...])][json_array],<br> [JSON_MERGE(json_doc, json_doc[, json_doc] ...)][json_merge],<br> [JSON_SET(json_doc, path, val[, path, val] ...)][json_set],<br> [JSON_INSERT(json_doc, path, val[, path, val] ...)][json_insert],<br> [JSON_REPLACE(json_doc, path, val[, path, val] ...)][json_replace],<br> [JSON_REMOVE(json_doc, path[, path] ...)][json_remove] |
+| [Date and Time Functions](/dev/reference/sql/functions-and-operators/date-and-time-functions.md) | [`DATE_FORMAT()`](https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_date-format)  |
+
+## Blacklist specific operators
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
## Blacklist specific expressions
```

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+If unexpected behavior occurs during the calculation of functions due to pushed down operators, you can quickly restore the execution by blacklisting specific operators. Specifically, you can prohibit operators from being pushed down by adding them to the blacklist `mysql.expr_pushdown_blacklist`.
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
If unexpected behavior occurs during the calculation of a function caused by its pushdown, you can quickly restore the application by blacklisting that function. Specifically, you can prohibit an expression from being pushed down by adding it to the `mysql.expr_pushdown_blacklist` blacklist.
```

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md (hunk 1 above)
Comment (anotherrachel) [reply dev/reference/sql/functions-and-operators/expressions-pushed-down.md]: > Can the `pushdown` be used as a noun？
> This document introduces which TiDB operators can be pushed down to TiKV, and how to disable this function by putting specific operators into a blacklist.

Thanks for your sharp eyes! I have done a little search actually, and to answer your question, yes, `pushdown` can be used as a noun, but with a totally different meaning. However, in [one of the blogs that we have posted](https://pingcap.com/blog/adding-built-in-functions-to-tikv/), we used expressions like "select a function for pushdown", so it is justifiable to use `pushdown` here. Similarily, "pushdown" can also be used as an adjective.
Also, apologies for my late follow-up.

**Comment by lilin90** [comment]
@anotherrachel Please address comments.

**Comment by anotherrachel** [comment]
> @anotherrachel Please address comments.

Actually, the comments are still under discussion. I'll update it when @yikeke and I reach the same conclusion.

**Comment by yikeke** [comment]
> > @anotherrachel Please address comments.
> 
> Actually, the comments are still under discussion. I'll update it when @yikeke and I reach the same conclusion.

Right. Will update my comments later.

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+# List of Operators for Pushdown
+
+When TiDB reads data from TiKV, TiDB tries to push down some expressions to TiKV to be processed thus reducing the amount of data transferred and the computational pressure on a single TiDB node. This document introduces operators that TiDB supports for pushdown and how to prohibit specific operators from being pushed down by adding them to the blacklist.
+
+## List of operators for pushdown
+
+| Type | Operators |
+| :-------------- | :------------------------------------- |
+| [Logical Functions](/dev/reference/sql/functions-and-operators/operators.md#logical-operators) | AND (&&), OR (&#124;&#124;), NOT (!) |
+| [Comparison Functions](/dev/reference/sql/functions-and-operators/operators.md#comparison-functions-and-operators) | <, <=, =, != (<>), >, >=, [`<=>`](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#operator_equal-to), [`IN()`](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_in), IS NULL, LIKE, IS TRUE, IS FALSE, [`COALESCE()`](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_coalesce) |
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
| [Comparison operations](/dev/reference/sql/functions-and-operators/operators.md#comparison-functions-and-operators) | <, <=, =, != (<>), >, >=, [`<=>`](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#operator_equal-to), [`IN()`](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_in), IS NULL, LIKE, IS TRUE, IS FALSE, [`COALESCE()`](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_coalesce) |
```
please update the following rows accordingly.

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+| [Date and Time Functions](/dev/reference/sql/functions-and-operators/date-and-time-functions.md) | [`DATE_FORMAT()`](https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_date-format)  |
+
+## Blacklist specific operators
+
+If unexpected behavior occurs during the calculation of functions due to pushed down operators, you can quickly restore the execution by blacklisting specific operators. Specifically, you can prohibit operators from being pushed down by adding them to the blacklist `mysql.expr_pushdown_blacklist`.
+
+### Add to the blacklist
+
+To add one or more operators to the blacklist, perform the following steps:
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
To add one or more functions or operators to the blacklist, perform the following steps:
```

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+
+1. Insert the corresponding operator name in `mysql.expr_pushdown_blacklist`.
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
1. Insert the function or operator name to `mysql.expr_pushdown_blacklist`.
```

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+
+2. Execute `admin reload expr_pushdown_blacklist;` command.
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
2. Execute the `admin reload expr_pushdown_blacklist;` command.
```

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+
+### Remove from the blacklist
+
+To remove one or more operators from the blacklist, perform the following steps:
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
To remove one or more functions or operators from the blacklist, perform the following steps:
```

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+
+1. Delete the corresponding operator name in `mysql.expr_pushdown_blacklist`.
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
1. Delete the function or operator name in `mysql.expr_pushdown_blacklist`.
```

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+
+2. Execute `admin reload expr_pushdown_blacklist;` command.
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
2. Execute the `admin reload expr_pushdown_blacklist;` command.
```

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 15]
```diff
@@ -0,0 +1,147 @@
...
+
+### Instance
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
### Blacklist usage examples
```

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 16]
```diff
@@ -0,0 +1,147 @@
...
+
+The following example demonstrates how to add `<` and `>` to the blacklist, then remove `>` from the blacklist.
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
The following example demonstrates how to add the `<` and `>` operators to the blacklist, then remove `>` from the blacklist.
```

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 17]
```diff
@@ -0,0 +1,147 @@
...
+
+Whether the blacklist is effective can be observed from the results returned by `EXPLAIN` statement (See [how to understand `EXPLAIN` results](/dev/reference/performance/understanding-the-query-execution-plan.md)).
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
You can see whether the blacklist takes effect by checking the results returned by `EXPLAIN` statement (See [Understanding `EXPLAIN` results](/dev/reference/performance/understanding-the-query-execution-plan.md)).
```

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 18]
```diff
@@ -0,0 +1,147 @@
...
+| └─TableReader_8       | 3333.33  | root | data:Selection_7                                           |
+|   └─Selection_7       | 3333.33  | cop  | gt(test.t.a, 2)                                            |
+|     └─TableScan_6     | 10000.00 | cop  | table:t, range:[-inf,+inf], keep order:false, stats:pseudo |
++-----------------------+----------+------+------------------------------------------------------------+
+4 rows in set (0.00 sec)
+```
+
+> **Note:**
+>
+> - `admin reload expr_pushdown_blacklist` only works for TiDB server that executes this SQL statement. To enable the blacklist in every TiDB server in the cluster, you need to execute the SQL statement on each TiDB server.
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
> - `admin reload expr_pushdown_blacklist` only takes effect on the TiDB server that executes this SQL statement. To make it apply to all TiDB servers, execute the SQL statement on each TiDB server.
```

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 19]
```diff
@@ -0,0 +1,147 @@
...
+> - The blacklist feature of pushed down operators is supported in v3.0.0 or later.
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
> - The feature of blacklisting specific expressions is supported in TiDB 3.0.0 or later versions.
```

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 20]
```diff
@@ -0,0 +1,147 @@
...
+> - Note that in v3.0.3 or earlier, some of the operators (such as ">", "+", "is null") are not supported to add to the blacklist. This means you have to use aliases (case-insensitive) instead, as shown in the following table:
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
> - TiDB 3.0.3 or earlier versions does not support adding some of the operators (such as ">", "+", "is null") to the blacklist by using their original names. You need to use their aliases (case-insensitive) instead, as shown in the following table:
```

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 21]
```diff
@@ -0,0 +1,147 @@
...
+>
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
```

Move this table out of the note.

File: dev/reference/sql/functions-and-operators/reference.md (hunk 3 above)
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/reference.md]:
```suggestion
You can write expressions using literals, column names, NULL, built-in functions, operators and so on. For expressions that TiDB supports pushing down to TiKV, see [List of Expressions for Pushdown](/dev/reference/sql/functions-and-operators/expressions-pushed-down.md).
```

File: dev/TOC.md [hunk 22]
```diff
@@ -214,6 +214,7 @@
       - [Window Functions](/dev/reference/sql/functions-and-operators/window-functions.md)
       - [Miscellaneous Functions](/dev/reference/sql/functions-and-operators/miscellaneous-functions.md)
       - [Precision Math](/dev/reference/sql/functions-and-operators/precision-math.md)
+      - [List of Operators for Pushdown](/dev/reference/sql/functions-and-operators/expressions-pushed-down.md)
```
Comment (yikeke) [inline dev/TOC.md]:
```suggestion
      - [List of Expressions for Pushdown](/dev/reference/sql/functions-and-operators/expressions-pushed-down.md)
```
Or `List of Operations for Pushdown`

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 23]
```diff
@@ -0,0 +1,147 @@
+---
+title: List of Operators for Pushdown
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]: ditto

File: dev/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 24]
```diff
@@ -0,0 +1,147 @@
...
+summary: This document introduces a list of operators that can be pushed down to TiKV, the related operations of which are also provided.
+category: reference
+---
+
+# List of Operators for Pushdown
```
Comment (yikeke) [inline dev/reference/sql/functions-and-operators/expressions-pushed-down.md]: ditto

**Comment by lilin90** [comment]
@yikeke PTAL again~ Thanks!

**Comment by lilin90** [comment]
@fzhedu PTAL again. Thanks!

**Pull request merged**